src/lib/mina_base/test/account/account.ml

```diff
@@ -0,0 +1,273 @@
+open Core_kernel
+open Currency
+open Mina_base
+open Mina_numbers
+open Unsigned
+
+let gen_timing = Account.gen_timing Balance.max_int
+
+let gen_timing_with_necessary_vesting =
+  let open Quickcheck in
+  let open Account.Timing.As_record in
+  Account.gen_timing Balance.max_int
+  |> Generator.filter ~f:(fun t ->
+         let open Balance in
+         t.initial_minimum_balance - t.cliff_amount
+         |> Option.value_map ~default:false ~f:(fun to_vest -> to_vest > zero) )
+
+let%test_module "Test account's timing." =
+  ( module struct
+    open Account.Timing.As_record
+
+    let min_balance_at_slot (t : Account.Timing.as_record) =
```

Comment:
These functions seem useful to add to the `Account.Timing.As_record` module directly.

Reply:
Sadly, cannot move them to `Account.Timing`, as it would create a dependency cycle between `account.ml` and `account_timing.ml`. Could move those functions to the former, but I'm not sure if it really makes sense…